sdk/python/feast/pyspark/abc.py

```diff
@@ -25,6 +35,85 @@ def get_id(self) -> str:
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def get_status(self) -> SparkJobStatus:
+        """
+        Job Status retrieval
+
+        :return: SparkJobStatus
+        """
+        raise NotImplementedError
+
+
+class SparkJobParameters(abc.ABC):
+    @abc.abstractmethod
+    def get_name(self) -> str:
+        """
+        Getter for job name
+        :return: Job name
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_main_file_path(self) -> str:
+        """
+        Getter for jar | python path
+        :return: Full path to file
+        """
+        raise NotImplementedError
+
+    def get_class_name(self) -> Optional[str]:
+        """
+        Getter for main class name if it's applicable
+        :return: java class path, e.g. feast.ingestion.IngestionJob
+        """
+        return None
+
+    @abc.abstractmethod
+    def get_arguments(self) -> List[str]:
+        """
+        Getter for job arguments
+        E.g., ["--source", '{"kafka":...}', ...]
+        :return: List of arguments
+        """
+        raise NotImplementedError
+
+
+class RetrievalJobParameters(SparkJobParameters):
+    def __init__(
+        self,
+        feature_tables: List[Dict],
+        feature_tables_sources: List[Dict],
+        entity_source: Dict,
+        destination: Dict,
+        **kwargs,
+    ):
+        self._feature_tables = feature_tables
+        self._feature_tables_sources = feature_tables_sources
+        self._entity_source = entity_source
+        self._destination = destination
+
+    def get_name(self) -> str:
```

Comment:
there is a chance that we'll run into limits on name length on different platforms, on EMR it is 256 chars max